pkg/scheduler/factory.go

```diff
@@ -539,8 +539,8 @@ func (c *Configurator) getPriorityConfigs(priorityKeys sets.String) ([]prioritie
 
 // GetPredicateMetadataProducer returns a function to build Predicate Metadata.
 // It is used by the scheduler and other components, such as k8s.io/autoscaler/cluster-autoscaler.
-func (c *Configurator) GetPredicateMetadataProducer() (predicates.PredicateMetadataProducer, error) {
-	return getPredicateMetadataProducer()
+func (c *Configurator) GetPredicateMetadataProducer() (predicates.MetadataProducer, error) {
+	return getPredicateMetadataProducer(*c.factoryArgs)
```

Comment:
is this derefrecning safe?

Reply:
also, you can get rid of this function completely, CA are moving to scheduler.New in 1.17

Reply:
It looks like there was no reason for it to be pointer anyways.
Function removed :)